src/CounterInterceptor/ClientCounter.cs

```diff
@@ -0,0 +1,89 @@
+namespace TemporalioSamples.CounterInterceptor;
+
+using System.Numerics;
+
+public class ClientCounter
+{
+    private const string NumberOfWorkflowExecutions = "numOfWorkflowExec";
+    private const string NumberOfSignals = "numOfSignals";
+    private const string NumberOfQueries = "numOfQueries";
+    private static Dictionary<string, Dictionary<string, BigInteger?>> perWorkflowIdDictionary =
```

Comment:
```suggestion
    private static Dictionary<string, WorkflowCounts> workflowCounts = new();
```

And have a mutable record like:

```public record WorkflowCounts
{
    uint64 Executions { get; internal set; }
    uint64 Signals { get; internal set; }
    uint64 Queries { get; internal set; }
}
```

Could add a `ToString()` on it if you must have multiline strings though may not be needed. And just call the field `workflowCounts`. You don't have to name the field by what the key is and what the value is. I haven't tested any of this, but can be cleaned up a bit

Reply:
Fixed. Added a bit of complexity in the Add() method but overall I like it.